doc/source/docs/reference.rst

```diff
@@ -1,11 +1,15 @@
 .. _reference:
 
-Reference
-=========
+API Reference
+=============
+
+The API Reference provides an overview of all public objects, functions and methods implemented in GeoPandas. All classes and function exposed in ``geopandas.*`` namespace plus those listed in the reference are public.
+
+.. warning::
+   The ``geopandas.array`` and ``geopandas.base`` modules are private. Stable functionality in such modules is not guaranteed.
```

Comment:
In one of our earlier discussion, we agreed that `GeometryArray` will not (yet) be in the API, so I am marking the module as private.